internal/server/http/server.go

```diff
@@ -76,28 +75,31 @@ func (s *Server) Build() error {
 	if _, err := os.Stat(s.impostersPath); os.IsNotExist(err) {
 		return fmt.Errorf("%w: the directory %s doesn't exists", err, s.impostersPath)
 	}
-	var imposterFileCh = make(chan string)
-	var done = make(chan bool)
+	done := make(chan struct{})
 
-	go func() {
-		findImposters(s.impostersPath, imposterFileCh)
-		done <- true
-	}()
-	for {
+	imposterFiles, errc := findImposters(s.impostersPath, done)
+	go s.processImposters(imposterFiles, done)
+
+	if err := <-errc; err != nil {
```

Comment:
May be we should do a error check before running goroutine? So there will be no post effects, if we will do something more complicated in future.

Reply:
That would mean waiting for `findImposters` to walk through all files before starting `processImposters` function. What I had in mind is that error during `findImposters` call would be serious enough to stop the server execution and as such we don't need to care about side effects. But I don't have strong opinion on that.